packages/react-dom/src/__tests__/ReactDOMFizzServer-test.js

```diff
@@ -135,14 +168,15 @@ describe('ReactDOMFizzServer', () => {
     while (fakeBody.firstChild) {
       const node = fakeBody.firstChild;
       if (
-        node.nodeName === 'SCRIPT' &&
+        (node.nodeName === 'SCRIPT' || node.nodeName === 'script') &&
         (CSPnonce === null || node.getAttribute('nonce') === CSPnonce)
       ) {
-        const script = document.createElement('script');
+        const script = document.createElement('SCRIPT');
```

Comment:
The branch here is really just assuming that these are inline scripts and to get them to execute, we need to reinsert them because (for security reasons innerHTML doesn't).

What you can do is check if `.src` is specified and if so, fake reading the source file as source, and then injecting it as a script tag with `textContent` instead. That would simulate more what the browser would do here.